pkg/csi/service/vanilla/controller.go

```diff
@@ -2510,7 +2511,21 @@ func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesReques
 				}
 				CNSVolumesforListVolume = cnsQueryResult.Volumes
 			}
+
+			// Get all nodes from the vanilla K8s cluster from the node manager
+			allNodeVMs, err := c.nodeMgr.GetAllNodes(ctx)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to get nodes(node vms) in the vanilla cluster. Error: %v", err)
+			}
+
+			volumeIDToNodeUUIDMap, err = getBlockVolumeIDToNodeUUIDMap(ctx, c, allNodeVMs)
```

Comment:
Can you add a comment here that we are getting this map once per resync cycle and it will be used later while processing the volumes.

Reply:
I have added it.